### PR TITLE
ARROW-4835: [GLib] Add boolean operations

### PR DESCRIPTION
**Pull request by kou**

### CODE REVIEW

File: cpp/src/arrow/compute/api.h [hunk 1]
```diff
@@ -21,7 +21,8 @@
 #include "arrow/compute/context.h"  // IWYU pragma: export
 #include "arrow/compute/kernel.h"   // IWYU pragma: export
 
-#include "arrow/compute/kernels/cast.h"  // IWYU pragma: export
-#include "arrow/compute/kernels/hash.h"  // IWYU pragma: export
+#include "arrow/compute/kernels/boolean.h" // IWYU pragma: export
```
Comment (kou) [inline cpp/src/arrow/compute/api.h]: @wesm @xhochy Can we export this?

Comment (xhochy) [reply cpp/src/arrow/compute/api.h]: I have no expericence with IWYU yet, so @wesm needs to answer.

Comment (wesm) [reply cpp/src/arrow/compute/api.h]: Yes, that's fine

Comment (kou) [reply cpp/src/arrow/compute/api.h]: OK. Thanks for confirming this!

**Review by shiro615 — APPROVED** [review]
LGTM

**Pull request closed without merging**